lightning/src/routing/router.rs

```diff
@@ -181,7 +265,26 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, targ
 		return Err(LightningError{err: "Cannot generate a route of more value than all existing satoshis".to_owned(), action: ErrorAction::IgnoreError});
 	}
 
-	// We do a dest-to-source Dijkstra's sorting by each node's distance from the destination
+	// The general route finding idea is the following:
+	// 1. Fill first/last hops communicated by the caller.
+	// 2. Construct and store a path.
+	//    Remember which channels were used and how much capacity they have available,
+	//    so that future paths don't rely on the same capacity.
```

Comment:
>Until when the future paths don't rely on the same capacity ? Success or failure of previous paths ?

Will try to clarify.

>A follow-up could be to isolate the live capacity-bookkeeping logic in its own component. It would be passed across call to get_route.

Perhaps, although it would be hard to keep that up-to-date. Definitely makes sense for our own channels.